sqs/src/test/scala/docs/scaladsl/SqsPublishSpec.scala

```diff
@@ -191,6 +198,30 @@ class SqsPublishSpec extends AnyFlatSpec with Matchers with DefaultTestContext w
     }
   }
 
+  it should "publish batch of SendMessageRequests with individual delays and pull them" taggedAs Integration in {
+    new IntegrationFixture {
+      // #batch-send-request
+      val messageDelaySeconds = 3
+      val messages = for (i <- 0 until 10)
+        yield
+          SendMessageRequest
+            .builder()
+            .messageBody(s"Message - $i")
+            .delaySeconds(messageDelaySeconds) // SQS queue should process messages after 5 seconds
+            .build()
+
+      val future = Source
+        .single(messages)
+        .runWith(SqsPublishSink.batchedMessageSink(queueUrl))
+      // #batch-send-request
+
+      future.futureValue shouldBe Done
+
+      receiveMessages(10) should have size 0 // delay hasn't elapsed
+      receiveMessages(10, messageDelaySeconds + 1) should have size 10 // delay has elapsed
```

Comment:
thanks - this looks good but I would be concerned that 3 seconds + 1 might be a bit brittle on GitHub CI infrastructure

I'm happy enough to merge as is and worry about refactoring further if we hit issues with the test failing intermittently.

I think I'll leave this open for a couple of days to let other people review it.

Reply:
@pjfanning are you concerned in general about this type of test which relies on time delays, or are you concerned about +1 specifically, in terms that it's too short?

Cause I can tune it immediatelly

Reply:
I'm possibly overthinking this but we need to find the right balance between having a test that is robust but that doesn't take excessively long to run. GitHub CI is a great resource but the VMs can run a bit slower and less uniformly than a PC. This is why I'm often suspicious when I see artbitrary values like 1 second.
If you have any ideas, try them out.
The ideal would be that for the test of the receiving the messages after the delay that we actively gather the messages and pass as soon as we have the 10. We can set some deadline and this can be generous. The idea is that usually, the 10 messages should arrive very quickly after the delay expires but that occasionally, we might need to wait - say maybe 10 seconds, just in case. It would not be ideal to always wait the extra 10 seconds.
I haven't run your test myself (yet) but may have time tonight to run it a few times to see how it works over a few runs.

Reply:
I understand what you mean.

In that case, the current setup should be good enough. By setting `waitTimeSeconds` in the `ReceiveMessageRequest`, we specify the **maximum** time which can elapse when waiting for the messages during the polling process. Also, the `get` method which is used on the future, which is returned by `awsSqsClient.receiveMessage`, has a timeout on itself, which should be larger than `waitTimeSeconds`, just to ensure that we won't be waiting for the future to complete indefinitely (however, this case shouldn't happen).

So, to summarize, we don't have a way of waiting for the delay to pass, unless we specify `waitTimeSeconds` in the request. `timeoutSeconds` is then just a precaution to ensure that we will eventually stop waiting for the future to complete.

The reason why I used `messageDelaySeconds + 1` in the method call is to ensure that we always pass a number which is larger than the delay (instead of using an arbitrary value like 60, or another variable, because it would need to have a value larger than `messageDelaySeconds`.

I can increase this arbitrary number to 5 or 10, just in case there is a hiccup in the test container for sqs, but technically, there can still be cases where this doesn't help either.

Let me know if this helps / clarifies anything or should I rephrase?

Reply:
I updated the `messageDelaySeconds` to be 5 seconds, and argument which is passed to `receiveMessages` is now 10 seconds in total (`waitTimeSeconds`). Timeout of the future (`timeoutSeconds`) will be set to 12 seconds.

But, if you run the test a couple of times, you'll notice that it finishes after just 5 seconds. Reason is that `waitTimeSeconds` is the maximum time that the client will spend waiting for the messages.